### DISCUSSION TITLE
Problems With the SnowflakeOperator.

### DISCUSSION THREAD

**Discussion by SlayerDevelopment** [apache/airflow#32957] [Q&A]
Hello Community, I need some advices and some answers why when I save my dag like this works:
![image](https://github.com/apache/airflow/assets/28604391/3620be1d-f4f0-4114-b041-500e27ff8165)
but if save like this doesnt work:
![image](https://github.com/apache/airflow/assets/28604391/f72e00f5-20c6-4965-911e-574ef3a84d50)
the dag_list list is instanced like this:
![image](https://github.com/apache/airflow/assets/28604391/c4e7660e-cd7e-474f-b688-7ce99ba35ecf)
What I mean with works I dont see the dags in the UI.

Please I need advices and answers.

**Comment by potiuk** [comment]
Comment (not sure if helpful and it might not be me who might be further involved)

But it seems like debugging exercise with your code. And it's next to impossible for human to do screenshot-debugging. People are not best in trying to compile/validate code in their heads from screenshots. I suggest that you make some debugging (like usual with Python code to see what's going on. - what objects get created etc. and explain what you have found. 

What could also be helpful if you produce a usable diff between the two "screenshots" explaining what changed between them.

Just a suggestion to make you more efficient in seeking for help - if you help others to help you, you are more likely to get help.

**Comment by SlayerDevelopment** [comment]
ok @potiuk yo are right, attach the screenshots, ok when I applied the first example of code show me all dags all task like I want to deploy and use it:
![image](https://github.com/apache/airflow/assets/28604391/7a7332d4-9a9a-4df5-b0bd-ea2314762213)
and the log scheduler show me the next:
![image](https://github.com/apache/airflow/assets/28604391/64fbb139-6dbb-4094-b801-4c782826cab6)
But when I use snowflakeoperators this happen in the UI:
![image](https://github.com/apache/airflow/assets/28604391/d6df7af9-6707-49c6-9222-d83a2bbe8c1f)
and the scheduler log show me this:
![image](https://github.com/apache/airflow/assets/28604391/8cb58aaf-a80b-4ccd-9f4a-8e5665519f33)
why this happen:

**Reply by potiuk** [reply]
Well you do have  "DAG import Errors" - so maybe that's the problem? Have you checked them?

**Reply by potiuk** [reply]
It's the big red thing that should drag your attention

**Reply by SlayerDevelopment** [reply]
Hello @potiuk no in this part there is nothing about the dag file where I'm working.

**Reply by potiuk** [reply]
Then no idea. Try to run `python your_dag.py` and debug it as a regular Python code and see what's different between the two.  There is also `airflow tasks test` that you might want to use to test your DAG.

When you do it, and show your effort, diffe and the things I explaied before then likely you will have some findings that will allow somoene to help you (Again not necessarily me).

**Reply by potiuk** [reply]
It's REALLY difficult to remotely debug somoene's code and even more difficult when this debugging is based on screenshots of the code. It's much easier for you to debug it and describe in text the results of your investigation.

**Reply by SlayerDevelopment** [reply]
Ok @potiuk you are right this is my code please review.
https://gist.github.com/SlayerDevelopment/c44a8c712c3506a5539a8e68089d4222

**Reply by potiuk** [reply]
No. you are wrong. I am mot going to review your code. 
What is the difference between the code you had working. What your investigation shows?

I am sorry but you completely miss the point

**Reply by SlayerDevelopment** [reply]
ok @potiuk thank you.
